src/extension/importer/rest/src/test/java/org/geoserver/importer/rest/RequestContextListener.java

```diff
@@ -0,0 +1,45 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.importer.rest;
+
+import org.geoserver.catalog.ResourceInfo;
+import org.geoserver.security.AbstractResourceAccessManager;
+import org.geoserver.security.DataAccessLimits;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * The propose of this class is to test that requests spring context are correctly shared
```

Comment:
Well it was the only way I found to test that the spring context was share between several threads.